conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/NegationRuleInterpreter.scala

```diff
@@ -32,28 +32,30 @@ case class NegationRuleInterpreter(rule: NegationConformanceRule) extends RuleIn
   override def conform(df: Dataset[Row])(implicit spark: SparkSession, dao: EnceladusDAO, progArgs: CmdConfig): Dataset[Row] = {
     NegationRuleInterpreter.validateInputField(progArgs.datasetName, df.schema, rule.inputColumn)
 
-    val fieldType = SchemaUtils.getFieldType(rule.inputColumn, df.schema).get
+    val field = SchemaUtils.getStructField(rule.inputColumn, df.schema).get
+
     val negationErrUdfCall = callUDF("confNegErr", lit(rule.outputColumn), col(rule.inputColumn))
     val errCol = "errCol"
 
-    fieldType match {
+    field.dataType match {
       case _: DecimalType =>
         // Negating decimal cannot fail
         DeepArrayTransformations.nestedWithColumnMap(df, rule.inputColumn, rule.outputColumn, c => negate(c))
       case _: BooleanType =>
+        // Negating Boolean cannot fail
         DeepArrayTransformations.nestedWithColumnMap(df, rule.inputColumn, rule.outputColumn, c => not(c))
       case _: DoubleType | _: FloatType =>
         // Negating floating point numbers cannot fail, but we need to account
         // for signed zeros (see the note for getNegator()).
-        DeepArrayTransformations.nestedWithColumnMap(df, rule.inputColumn, rule.outputColumn, c => getNegator(c, fieldType))
+        DeepArrayTransformations.nestedWithColumnMap(df, rule.inputColumn, rule.outputColumn, c => getNegator(c, field))
       case _ =>
```

Comment:
Since there are already changes I would go a step further and on line `51` do `case dataType =>` and then use the `dataType` value on line `54`

Reply:
OK